src/core/raster/qgsrasterdrawer.h

```diff
@@ -42,8 +42,9 @@ class CORE_EXPORT QgsRasterDrawer
      * The QgsRasterDrawer constructor.
      * \param iterator the raster iterator to fetch data from
      * \param dpiTarget the target \a dpi (dots per inch) to be taken into consideration when rendering
+     * \param scaleFactor the scale factor for the render to convert painter units to physical sizes
      */
-    QgsRasterDrawer( QgsRasterIterator *iterator, double dpiTarget = -1.0 );
+    QgsRasterDrawer( QgsRasterIterator *iterator, double dpiTarget = -1.0, double scaleFactor = -1.0 );
```

Comment:
It would be cleaner at this stage to pass on the whole render context to QgsRasterDrawer. How do you feel about deprecating this constructor and making a new one:

    QgsRasterDrawer( QgsRasterIterator *iterator );

and then deprecating the current draw method and making one:

        void draw( QgsRenderContext& context, QgsRasterViewPort *viewPort, QgsRasterBlockFeedback *feedback = nullptr );

Reply:
Sounds good to me @nyalldawson